gensim/models/tfidfmodel.py

```diff
@@ -29,7 +31,7 @@ def resolve_weights(smartirs):
 
     Parameters
     ----------
-    smartirs : str
+    smartirs : str or None
```

Comment:
What does `None` do?

Reply:
Confuse the reader. I removed the polymorphism in 35f0f9db.